src/sign-authorization.ts

```diff
@@ -0,0 +1,127 @@
+import { encode } from '@ethereumjs/rlp';
+import { ecsign, publicToAddress, toBuffer } from '@ethereumjs/util';
+import { bytesToHex } from '@metamask/utils';
+import { keccak256 } from 'ethereum-cryptography/keccak';
+
+import { concatSig, isNullish, recoverPublicKey } from './utils';
+
+/**
+ * The authorization struct as defined in EIP-7702.
+ *
+ * @property chainId - The chain ID or 0 for any chain.
+ * @property contractAddress - The address of the contract being authorized.
+ * @property nonce - The nonce of the signing account (at the time of submission).
+ */
+export type Authorization = [
```

Comment:
Minor, could `Authorization` be or become too generic a term in the context of signing?

Should we call this something like `EIP7702Authorization`?
Maybe `7702` as the filename?

Reply:
Yeah, I think that makes sense - it's already a bit of an overloaded term.

Reply:
I named the file `sign-eip7702-authorization.ts` for a combination of clarity and consistency with the other signing files.